src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/LPG2RDFConfigurationReader.java

```diff
@@ -330,7 +330,145 @@ public EdgeLabelMapping getEdgeLabelMapping(final Model lpg2Rdf, final Resource
 
         return createEdgeLabelMapping(edgeLabelMappingResource, edgeLabelMappingResourceType);
     }
+    public PropertyNameMapping createIRIBasedPropertyNameMapping(final Resource propertyNameMappingResource){
+        final StmtIterator prefixOfIRIsIterator = propertyNameMappingResource.listProperties(LPG2RDF.prefixOfIRIs);
+        if(!prefixOfIRIsIterator.hasNext()){
+            throw new IllegalArgumentException("prefixOfIRIs is required!");
+        }
+        final RDFNode prefixOfIRIObj = prefixOfIRIsIterator.next().getObject();
+        if(prefixOfIRIsIterator.hasNext()){
+            throw new IllegalArgumentException("An instance of IRIBasedPropertyNameMapping has more than one prefixOfIRIs property!");
+        }
+
+        if (!prefixOfIRIObj.isLiteral() || !prefixOfIRIObj.asLiteral().getDatatypeURI().equals(XSD.anyURI.getURI())){
+            throw new IllegalArgumentException("prefixOfIRIs is invalid, it should be a xsd:anyURI!");
+        }
+        final String prefixOfIRIUri = prefixOfIRIObj.asLiteral().getString();
+        try{
+            return new PropertyNameMappingToURIsImpl(URI.create(prefixOfIRIUri).toString());
+        }
+        catch (IllegalArgumentException exception){
+            throw new IllegalArgumentException("prefixOfIRIs is an invalid URI!");
```

Comment:
If a developer gets this error message for an LPG-to-RDF description that contains several `lr:prefixOfIRIs` triples, he/she would not know which URI this message refers to. Therefore, better be explicit:
```suggestion
            throw new IllegalArgumentException("prefixOfIRIs ( + prefixOfIRIUri + ) is an invalid URI!");
```

Reply:
Please also double check the other error messages throughout this PR for similar issues.